src/common/async.test.ts

```diff
@@ -0,0 +1,17 @@
+import * as fc from 'fast-check'
+import { race } from './async'
```

Comment:
Wait. Why do we have our own implementation for [`Promise.race`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race)?

No action required immediately, but I think we should switch that out for the native variant. No need to test this ourselves then :+1:

Reply:
Great point, I was also noticing that Jest coverage doesn't think we are actually calling `promise.then(resolve, reject)` in the implementation. (I tested it with logging and we are, but it's not ideal that it doesn't show up in coverage.)

Reply:
TIL! I didn't realize Promises has `.race` built in.